cpp/cgi_progs/newsletter_uploader.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+import cgi
+import cgitb
+import os
+
+
+cgitb.enable(display=0, logdir="/usr/local/var/log/tuefind/")
+form = cgi.FieldStorage()
+
+
+# A nested FieldStorage instance holds the file
+file_field = form['file']
+
+
+# Test if the file was uploaded
+if file_field:
+    # Strip leading path from file name to avoid directory traversal attacks
+    filename = os.path.basename(file_field.filename)
+    open("/usr/local/var/lib/tuelib/newsletters" + filename, 'w').write(file_field.file.read())
```

Comment:
What if newsletters subdir doesn't exist yet? (create dynamically, or let installer create it?)

Reply:
Installer does it now.